cmd/go-sbot/config.go

```diff
@@ -112,48 +118,98 @@ func ReadEnvironmentVariables(config *SbotConfig) {
 		config.Hops = uint(hops)
 		config.presence["hops"] = true
 	}
+
 	if val := os.Getenv("SSB_MUXRPC_ADDRESS"); val != "" {
 		config.MuxRPCAddress = val
 		config.presence["lis"] = true
 	}
+
 	if val := os.Getenv("SSB_WS_ADDRESS"); val != "" {
 		config.WebsocketAddress = val
 		config.presence["wslis"] = true
 	}
+
 	if val := os.Getenv("SSB_EBT_ENABLED"); val != "" {
-		config.EnableEBT = BooleanIsTrue(val)
+		config.EnableEBT = readEnvironmentBoolean(val)
 		config.presence["enable-ebt"] = true
 	}
+
 	if val := os.Getenv("SSB_CONN_FIREWALL_ENABLED"); val != "" {
-		config.EnableFirewall = BooleanIsTrue(val)
+		config.EnableFirewall = readEnvironmentBoolean(val)
 		config.presence["promisc"] = true
 	}
+
 	if val := os.Getenv("SSB_SOCKET_ENABLED"); val != "" {
-		config.NoUnixSocket = !BooleanIsTrue(val)
+		config.NoUnixSocket = !readEnvironmentBoolean(val)
 		config.presence["nounixsock"] = true
 	}
+
 	if val := os.Getenv("SSB_CONN_DISCOVERY_UDP_ENABLED"); val != "" {
-		config.EnableDiscoveryUDP = BooleanIsTrue(val)
+		config.EnableDiscoveryUDP = readEnvironmentBoolean(val)
 		config.presence["localdiscov"] = true
 	}
+
 	if val := os.Getenv("SSB_CONN_BROADCAST_UDP_ENABLED"); val != "" {
-		config.EnableAdvertiseUDP = BooleanIsTrue(val)
+		config.EnableAdvertiseUDP = readEnvironmentBoolean(val)
 		config.presence["localadv"] = true
 	}
+
 	if val := os.Getenv("SSB_CAP_SHS_KEY"); val != "" {
 		config.ShsCap = val
 		config.presence["shscap"] = true
 	}
+
 	if val := os.Getenv("SSB_CAP_HMAC_KEY"); val != "" {
 		config.Hmac = val
 		config.presence["hmac"] = true
 	}
 }
 
+func (booly ConfigBool) MarshalJSON() ([]byte, error) {
+	temp := (bool)(booly)
+	b, err := json.Marshal(temp)
+	return b, err
+}
+
+func (booly *ConfigBool) UnmarshalJSON(b []byte) error {
+	// unmarshal into interface{} first, as a bool can't be unmarshaled into a string
+	var v interface{}
+	err := json.Unmarshal(b, &v)
+	if err != nil {
+		return eout(err, "unmarshal config bool")
+	}
+
+	// go through a type assertion dance, capturing the two cases:
+	// 1. if the config value is a proper boolean, and
+	// 2. if the config value is a boolish string (e.g. "true" or "1")
+	var temp bool
+	if val, ok := v.(bool); ok {
+		temp = val
+	} else if s, ok := v.(string); ok {
+		temp = BooleanIsTrue(s)
+		if !temp {
+			// catch strings that cause a false value, but which aren't boolish
+			if s != "false" && s != "0" && s != "no" && s != "off" {
+				return errors.New("non-boolean string found when unmarshaling boolish values")
+			}
+		}
+	}
+	*booly = (ConfigBool)(temp)
+
+	return nil
+}
+
 func BooleanIsTrue(s string) bool {
```

Comment:
```suggestion
func booleanIsTrue(s string) bool {
```

only export whats necessary. Very unusual for a `package main`.

Reply:
yes true!